test/integration/virtual-rules/aria-text.js

```diff
@@ -0,0 +1,198 @@
+describe('aria-text virtual-rule', function () {
+  it('should incomplete for element with undefined children', function () {
+    var node = new axe.SerialVirtualNode({
+      nodeName: 'div',
+      attributes: {
+        role: 'text'
+      }
+    });
+
+    var results = axe.runVirtualRule('aria-text', node);
+
+    assert.lengthOf(results.passes, 0);
+    assert.lengthOf(results.violations, 0);
+    assert.lengthOf(results.incomplete, 1);
+  });
+
+  it('should fail for focusable widget children', function () {
+    var node = new axe.SerialVirtualNode({
+      nodeName: 'div',
+      attributes: {
+        role: 'text'
+      }
+    });
+    var child = new axe.SerialVirtualNode({
+      nodeName: 'button',
+      attributes: {
+        role: 'widget'
+      }
```

Comment:
Did you maybe mean something like this?

```suggestion
      nodeName: 'div',
      attributes: {
        role: 'button'
      }
```

role=widget doesn't do anything, and putting it on a button makes this basically the same as your next test.